raft/server.cc

```diff
@@ -1150,7 +1151,19 @@ future<> server_impl::read_barrier(seastar::abort_source* as) {
             leader = _fsm->current_leader();
         } else {
             auto applied = _applied_idx;
-            auto res = co_await get_read_idx(leader, as);
+            read_barrier_reply res;
+            bool need_retry = false;
+            try {
+                res = co_await get_read_idx(leader, as);
+            } catch (const seastar::rpc::closed_error& e) {
```

Comment:
Isn't this error specific to the RPC implementation we're using?
Another implementation of `raft::rpc` may use completely different exception types.
cc @gleb-cloudius WDYT?

Reply:
this is a temporary solution, I'll handle it later

Reply:
> Isn't this error specific to the RPC implementation we're using? Another implementation of `raft::rpc` may use completely different exception types. cc @gleb-cloudius WDYT?

This is true. The solution may be to define raft::intermittent_connection_error and require rpc implementation return it in case of connection errors.

Reply:
fixed